build/resolve_buck.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import os
+import platform
+import stat
+import subprocess
+import sys
+import urllib.request
+
+from pathlib import Path
+
+import buck_util
+
+
+# Target buck2 version strings, as reported by buck2 --version.
+# Note that the version string differs by platform.
+BUCK_VERSION_DATE = "2024-02-15"
+BUCK_VERSION_LINUX_GNU_X86 = "38f7c508bf1b87bcdc816bf56d1b9f2d2411c6be"
+BUCK_VERSION_DARWIN_ARM64 = "99e407b49dc432eda0cbddd67ea78346"
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Locates or downloads the appropriate version of buck2.",
+    )
+    parser.add_argument(
+        "--buck2",
+        default="",
+        help="Explicit 'buck2' path to use.",
+    )
+    parser.add_argument(
+        "--cache_dir",
+        help="Directory to cache downloaded versions of buck2.",
+    )
+    return parser.parse_args()
+
+
+def resolve_buck2(args: argparse.Namespace) -> str:
+    # Find buck2, in order of priority:
+    #  1) Explicit buck2 argument
+    #  2) System buck2 (if correct version)
+    #  3) Download buck2
+
+    # Determine the target buck2 version string according to the current
+    # platform. If the platform isn't linux or darwin, we won't perform
+    # any version validation.
+    target_buck_version = ""
+    buck_archive_name = ""
+
+    machine = platform.machine().lower()
+    is_x86 = machine == "x86" or machine == "x86_64" or machine == "amd64"
+    is_arm64 = machine == "arm64"
+
+    if sys.platform.startswith("linux") and is_x86:
+        buck_archive_name = "buck2-x86_64-unknown-linux-gnu.zst"
```

Comment:
`buck2-x86_64-unknown-linux-musl.zst` would be compatible with more linux systems, because it doesn't depend on a particular version of glibc.